File: CPP/program/Tabel.cpp
#include <iostream>
#include <string>

using namespace std;

//Kelas Tabel
class Tabel{

    //Atribut
    private:
    int baris;
    int kolom;

    //Constructor
    public:
    
    Tabel(){
        this->baris = 0;
        this->kolom = 0;
    }
    
    //Constructor
    Tabel(int baris, int kolom){
        this->baris = baris;
        this->kolom = kolom;
    }

    //Setter dan Getter
    int getBaris(){
        return this->baris;
    }

    void setBaris(int baris) {
        this->baris = baris;
    }
    
    int getKolom(){
        return this->kolom;
    }
    
    void setKolom(int kolom) {
        this->kolom = kolom;
    }

    //method membuat tabel
    //pada parameter isitabel isi panjang kolom sebanyak kolom
    void buatTabel(int n_baris, int n_kolom, string isiTabel[][9], string headTabel[]){
        int kolom=0, baris=0;
        int maxPerKolom[n_kolom];   //array of int menampung max char per kolom nya

        //mencari maxperkolom
        for(kolom = 0; kolom < n_kolom; kolom++){
            maxPerKolom[kolom] = headTabel[kolom].length();
            for(baris = 0; baris < n_baris; baris++){
                if(isiTabel[baris][kolom].length() > maxPerKolom[kolom]){
                    maxPerKolom[kolom] = isiTabel[baris][kolom].length();
                }
            }
            maxPerKolom[kolom]+=1;
        }

        cout << '\n';
        //buat garis
        pemisahBaris(n_kolom, maxPerKolom, '-');
        
        //header tabel
        for(kolom = 0; kolom < n_kolom; kolom++){
            cout << "| " + headTabel[kolom];
            if(headTabel[kolom].length() < maxPerKolom[kolom]){
                for(int spasi = 0; spasi < maxPerKolom[kolom]-headTabel[kolom].length(); spasi++){
                    cout << " ";
                }
            }
        }
        cout << "|\n";

        //buat garis
        pemisahBaris(n_kolom, maxPerKolom, '=');
        
        //isi tabel
        for(baris = 0; baris < n_baris; baris++){
            for(kolom = 0; kolom < n_kolom; kolom++){
                cout << "| " + isiTabel[baris][kolom];
                
                if(headTabel[kolom].length() < maxPerKolom[kolom]){
                    for(int spasi = 0; spasi < maxPerKolom[kolom]-isiTabel[baris][kolom].length(); spasi++){
                        cout << " ";
                    }
                }
            }
            cout << "|\n";
            //buat garis
            pemisahBaris(n_kolom, maxPerKolom, '-');
        }
    }

    void buatTabel6(int n_baris, int n_kolom, string isiTabel[][6], string headTabel[]){
        int kolom=0, baris=0;
        int maxPerKolom[n_kolom];   //array of int menampung max char per kolom nya

        //mencari maxperkolom
        for(kolom = 0; kolom < n_kolom; kolom++){
            maxPerKolom[kolom] = headTabel[kolom].length();
            for(baris = 0; baris < n_baris; baris++){
                if(isiTabel[baris][kolom].length() > maxPerKolom[kolom]){
                    maxPerKolom[kolom] = isiTabel[baris][kolom].length();
                }
            }
            maxPerKolom[kolom]+=1;
        }

        cout << '\n';
        //buat garis
        pemisahBaris(n_kolom, maxPerKolom, '-');
        
        //header tabel
        for(kolom = 0; kolom < n_kolom; kolom++){
            cout << "| " + headTabel[kolom];
            if(headTabel[kolom].length() < maxPerKolom[kolom]){
                for(int spasi = 0; spasi < maxPerKolom[kolom]-headTabel[kolom].length(); spasi++){
                    cout << " ";
                }
            }
        }
        cout << "|\n";

        //buat garis
        pemisahBaris(n_kolom, maxPerKolom, '=');
        
        //isi tabel
        for(baris = 0; baris < n_baris; baris++){
            for(kolom = 0; kolom < n_kolom; kolom++){
                cout << "| " + isiTabel[baris][kolom];
                
                if(headTabel[kolom].length() < maxPerKolom[kolom]){
                    for(int spasi = 0; spasi < maxPerKolom[kolom]-isiTabel[baris][kolom].length(); spasi++){
                        cout << " ";
                    }
                }
            }
            cout << "|\n";
            //buat garis
            pemisahBaris(n_kolom, maxPerKolom, '-');
        }
    }

    void buatTabel8(int n_baris, int n_kolom, string isiTabel[][8], string headTabel[]){
        int kolom=0, baris=0;
        int maxPerKolom[n_kolom];   //array of int menampung max char per kolom nya

        //mencari maxperkolom
        for(kolom = 0; kolom < n_kolom; kolom++){
            maxPerKolom[kolom] = headTabel[kolom].length();
            for(baris = 0; baris < n_baris; baris++){
                if(isiTabel[baris][kolom].length() > maxPerKolom[kolom]){
                    maxPerKolom[kolom] = isiTabel[baris][kolom].length();
                }
            }
            maxPerKolom[kolom]+=1;
        }

        cout << '\n';
        //buat garis
        pemisahBaris(n_kolom, maxPerKolom, '-');
        
        //header tabel
        for(kolom = 0; kolom < n_kolom; kolom++){
            cout << "| " + headTabel[kolom];
            if(headTabel[kolom].length() < maxPerKolom[kolom]){
                for(int spasi = 0; spasi < maxPerKolom[kolom]-headTabel[kolom].length(); spasi++){
                    cout << " ";
                }
            }
        }
        cout << "|\n";

        //buat garis
        pemisahBaris(n_kolom, maxPerKolom, '=');
        
        //isi tabel
        for(baris = 0; baris < n_baris; baris++){
            for(kolom = 0; kolom < n_kolom; kolom++){
                cout << "| " + isiTabel[baris][kolom];
                
                if(headTabel[kolom].length() < maxPerKolom[kolom]){
                    for(int spasi = 0; spasi < maxPerKolom[kolom]-isiTabel[baris][kolom].length(); spasi++){
                        cout << " ";
                    }
                }
            }
            cout << "|\n";
            //buat garis
            pemisahBaris(n_kolom, maxPerKolom, '-');
        }
    }
    
    //method untuk garis pemisah baris
    void pemisahBaris(int n_kolom, int maxPerKolom[], char simbol){
        for(int kolom = 0; kolom < n_kolom; kolom++){
            for(int perKolom = 0; perKolom < maxPerKolom[kolom]; perKolom++){
                cout << simbol;
            }
            cout << simbol;
            cout << simbol;

        }
        cout << simbol << '\n';
    }

    ~Tabel(){
            
    }
    
};

File: Python/program/Tabel.py
# Kelas Tabel
class Tabel:
    # Atribut
    __baris = 0
    __kolom = 0

    # Constructor
    def __init__(self, baris = "", kolom = ""):
        self.__baris = baris
        self.__kolom = kolom

    # Setter dan Getter
    def get_baris(self):
        return self.__baris
    
    def set_baris(self, baris):
        self.__baris = baris

    def get_kolom(self):
        return self.__kolom

    def set_kolom(self, kolom):
        self.__kolom = kolom

    # Method membuat tabel
    def buat_tabel(self, n_baris, n_kolom, isiTabel, headTabel):
        maxPerKolom = []  #Menampung max tiap kolom

        for index in range(n_kolom):
            maxPerKolom.append(0)

        # Mencari maxperkolom
        for kolom in range(n_kolom):
            maxPerKolom[kolom] = len(headTabel[kolom])
            for baris in range(n_baris):
                if(len(isiTabel[baris][kolom]) > maxPerKolom[kolom]):
                    maxPerKolom[kolom] = len(isiTabel[baris][kolom])
            maxPerKolom[kolom] += 1
        
        # buat garis pemisah baris
        self.pemisah_baris(n_kolom, maxPerKolom, '=')

        #Header tabel
        for kolom in range(n_kolom):
            print("| " + headTabel[kolom], end = "")
            if(len(headTabel[kolom]) < maxPerKolom[kolom]):
                for spasi in range(maxPerKolom[kolom]-len(headTabel[kolom])):
                    print(" ", end = "")
        print("|")

        # buat garis pemisah baris
        self.pemisah_baris(n_kolom, maxPerKolom, '=')

        #isi tabel
        for baris in range(n_baris):
            for kolom in range(n_kolom):
                print("| " + isiTabel[baris][kolom], end = "")

                if(len(headTabel[kolom]) < maxPerKolom[kolom]):
                    for spasi in range(maxPerKolom[kolom]-len(isiTabel[baris][kolom])):
                        print(" ", end = "")
            print("|")
            # buat garis pemisah baris
            self.pemisah_baris(n_kolom, maxPerKolom, '-')
                
    #Method membuat garis pemisah baris
    def pemisah_baris(self, n_kolom, maxPerKolom, simbol = ''):
        for kolom in range(n_kolom):
            for perkolom in range(maxPerKolom[kolom]):
                print(simbol, end='')
            print(simbol, end='')
            print(simbol, end='')
        print(simbol)